app/controllers/users_controller.rb

```diff
@@ -344,6 +345,46 @@ def logon_after_password_reset
     @success = I18n.t(message)
   end
 
+  def admin_login
+
+    unless SiteSetting.enable_sso && !current_user
+      return redirect_to path("/")
+    end
+
+    if request.put?
+      RateLimiter.new(nil, "admin-login-hr-#{request.remote_ip}", 6, 1.hour).performed!
+      RateLimiter.new(nil, "admin-login-min-#{request.remote_ip}", 3, 1.minute).performed!
+
+      user = User.find_by(email: params[:email])
+      if user && user.admin? && user.id != Discourse::SYSTEM_USER_ID
+        email_token = user.email_tokens.create(email: user.email)
+        Jobs.enqueue(:user_email, type: :admin_login, user_id: user.id, email_token: email_token.token)
+        @message = I18n.t("admin_login.success")
+      else
+        @message = I18n.t("admin_login.error")
+      end
+    elsif !params[:token].blank?
+      # token recieved, try to login
+      if EmailToken.valid_token_format?(params[:token])
+        @user = EmailToken.confirm(params[:token])
```

Comment:
Not exactly, this method confirms any token *generated via* `EmailToken` class.

In case of SSO enabled instance *the only way* to generate token via `EmailToken` class is using this method, and only Admin can do it.